lib/h3core.js

```diff
@@ -245,12 +245,15 @@ function destroyGeoPolygon(geoPolygon) {
     const geofenceOffset = 0;
     const numHolesOffset = geofenceOffset + SZ_GEOFENCE;
     const holesOffset = numHolesOffset + SZ_INT;
-    // Free the outer loop
-    C._free(C.getValue(geoPolygon + geofenceOffset, 'i8*'));
-    // Free the holes, if any
+    // Offset of the geofence vertex array pointer within the Geofence struct
+    const geofenceArrayOffset = SZ_INT;
+    // Free the outer vertex array
+    C._free(C.getValue(geoPolygon + geofenceOffset + geofenceArrayOffset, 'i8*'));
```

Comment:
This was the main issue, including the `geofenceArrayOffset` when looking up the vertex array pointer.